middleware/tracing.go

```diff
@@ -0,0 +1,49 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+)
+
+var (
+	TracingCtxKey       = &contextKey{"TracingHeaders"}
+	defaultTraceHeaders = []string{
+		http.CanonicalHeaderKey("X-Cloud-Trace-Context"), // GCP load balancer trace id
+		http.CanonicalHeaderKey("X-Amzn-Trace-Id"),       // AWS X-ray
+		http.CanonicalHeaderKey("CF-ray"),                // Cloudflare ray id
+		http.CanonicalHeaderKey("TraceId"),               // https://github.com/go-chi/traceid
+	}
+)
+
+func Tracing(traceHeaders ...string) func(next http.Handler) http.Handler {
+	if len(traceHeaders) == 0 {
+		traceHeaders = defaultTraceHeaders
+	}
```

Comment:
I would not put any defaults here but rather export the `defaultTraceHeaders` so from caller perspective it is clearer what is being done.